### ISSUE TITLE
Approve ERC20 token problem with metamask

### ISSUE THREAD

**Issue by AnhNguyenDuc2** [ProjectOpenSea/seaport-js#215]
function buildTransaction in src/utils/usecase.ts attach domain string using getTagFromDomain function and when dmain string is empy the domain still attached hashed empty string to calldata cause the trouble when calling approval from Metamask. 

Metamask throw : " Invalid token value; should be exactly 64 hex digits long (u256)"

since Metamask check the approval number string tobe exact 64 character.

**Comment by MiguelGonzales-WIV** [comment]
Same problem!

**Comment by gabspeck** [comment]
@MiguelGonzales-WIV if it's of interest to you, I managed to work around this by forking the library and adding a conditional around the code that appends the custom "domain" hash to the transaction data: https://github.com/ProjectOpenSea/seaport-js/commit/2250434643766ee52f778a61548c6064362bd737

**Comment by MiguelGonzales-WIV** [comment]
Seems it being added to the transaction data for other methods (e.g. fulfillOrder, fulfillOrders, and matchOrders). Has this been working for you for createOrder, where nature of order is an offer (i.e. offer property value is an ERC20 token)?
```
createOrder({
	offer: [
		// buyer's offer
		{
			token: '0xc1d6ef502ac5410b3f3706beb6a0808131337fb6',
			amount: offerPrice,
		}
	],
        ...rest of code
```

createOrder only works for ERC721 for me.

**Comment by ryanio** [comment]
This should be resolved now with [v1.2.1](https://github.com/ProjectOpenSea/seaport-js/releases/tag/v1.2.1)
